Sources/SwiftDocCUtilities/Action/Actions/Init/InitAction.swift

```diff
@@ -0,0 +1,155 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SwiftDocC
+
+/// An action that generates a documentation catalog from a template seed.
+public struct InitAction: Action {
+    
+    enum Error: DescribedError {
+        case catalogAlreadyExists, cannotCreate
+        var errorDescription: String {
+            switch self {
+                case .catalogAlreadyExists: return "A documentation catalog with the same name already exists in the output directory."
+                case .cannotCreate: return "Unable to generate the documentation catalog."
+            }
+        }
+    }
+    
+    private let catalogOutputURL: URL
+    private let catalogTemplateKind: CatalogTemplateKind
+    private let diagnosticEngine: DiagnosticEngine = DiagnosticEngine(treatWarningsAsErrors: false)
+    private let documentationTitle: String
+    private var resourceDocumentationLink: String {
+        switch catalogTemplateKind {
+        case .articleOnly:
+            "https://www.swift.org/documentation/docc/"
+        case .tutorial:
+            "https://www.swift.org/documentation/docc/tutorial-syntax"
```

Comment:
Using `switch` as an expression is new in Swift 5.9. We still use Swift 5.6

```suggestion
            return "https://www.swift.org/documentation/docc/"
        case .tutorial:
            return "https://www.swift.org/documentation/docc/tutorial-syntax"
```